### DISCUSSION TITLE
Running keystone with NodeJS flags

### DISCUSSION THREAD

**Discussion by Arthur-Ruthenberg** [keystonejs/keystone#8972] [Questions] [answered]
Hi,
Is there a way to run keystone with NodeJS flags?
For example, I am trying to do a memory trace, but I can't seem to find a way to run keystone with the --trace flag.
Anyone found a way to do it?

Thanks!

**Comment by unrevised6419** [comment, marked as answer]
You can run directly Keyston CLI from `.bin` folder

```bash
node --trace node_modules/.bin/keystone
```

Or you can import `@keystone-6/core/scripts`, which will do the same as running the Keystone CLI.

```bash
node --trace -e "require('@keystone-6/core/scripts')"
```

**Reply by Arthur-Ruthenberg** [reply]
@iamandrewluca thank you!!!

**Reply by unrevised6419** [reply]
@Arthur-Ruthenberg I updated my answer with a more straight forward solution.

**Reply by Arthur-Ruthenberg** [reply]
I see. great. And if I want to run other scripts, from the package-json?  how can that be achieved?

**Reply by unrevised6419** [reply]
Any NPM binary you run from npm scripts exists in `node_modules/.bin`, if that file is a JS file, it can be run with node directly.
